File: src/main/java/me/lokka30/levelledmobs/rules/MergableRule.java
package me.lokka30.levelledmobs.rules;

import javax.annotation.Nullable;

/**
 * Provides a common interface for various rules
 * that can be merged together
 *
 * @author stumper66
 * @since 3.12.0
 */
public interface MergableRule {
    void merge(final @Nullable MergableRule mergableRule);

    boolean doMerge();

    Object cloneItem();
}

File: levelledmobs-plugin/src/main/kotlin/io/github/arcaneplugins/levelledmobs/annotations/MergableRule.kt
package io.github.arcaneplugins.levelledmobs.annotations

/**
 * Provides a common interface for various rules
 * that can be merged together
 *
 * @author stumper66
 * @since 3.12.0
 */
interface MergableRule {
    fun merge(mergableRule: MergableRule?)

    val doMerge: Boolean

    fun cloneItem(): Any
}